drivers/hmis/app/graphql/mutations/submit_form.rb

```diff
@@ -52,16 +55,33 @@ def resolve(input:)
       # Run processor to create/update record(s)
       custom_form.form_processor.run!
 
+      # Run custom validator for any warnings/errors (like closing a Project)
+      validator = klass.validators.find { |v| v.respond_to?(:hmis_validate) }
+      if validator.present?
+        validation_errors = validator.hmis_validate(record, ignore_warnings: input.confirmed)
+        errors.push(*validation_errors)
+      end
+      return { errors: errors } if errors.any?
+
       # Run both validations
       record_valid = record.valid?
       custom_form_valid = custom_form.valid?
 
       if record_valid && custom_form_valid
-        # Save CustomForm to save any related records (if any)
-        custom_form.save!
-        # Save the record
-        record.save!
-        record.touch
+        # Perform any side effects
+        perform_side_effects(record)
+
+        if record.is_a? Hmis::Hud::HmisService
+          record.owner.save! # Save the actual service record
+          record = Hmis::Hud::HmisService.find_by(owner: record.owner) # Refresh from View
+          custom_form.owner = record # Set owner_id to the View id
```

Comment:
I feel like we shouldn't do this (have another table reference the computed HmisService view ID). The CustomForm owner should probably be the actual record (Service or HudService)